tests/Unit/FluidXPlateTest.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mll\LiquidHandlingRobotics\Tests\Unit;
+
+use Mll\LiquidHandlingRobotics\FluidXPlate;
+use Mll\LiquidHandlingRobotics\TecanScanException;
+use Mll\Microplate\Coordinate;
+use Mll\Microplate\CoordinateSystem96Well;
+use PHPUnit\Framework;
+use TypeError;
+
+class FluidXPlateTest extends Framework\TestCase
+{
+    public function testCreateFromStringEmpty(): void
+    {
+        $this->expectExceptionObject(new TecanScanException('Invalid rack id:'));
+        new FluidXPlate('');
+    }
+
+    public function testCreateWithRandomNameAndReturnsIt(): void
+    {
+        $rackId = 'testInvalidRackId';
+        $this->expectExceptionObject(new TecanScanException('Invalid rack id:' . $rackId));
+        new FluidXPlate($rackId);
+    }
+
+    public function testCreatesSuccessfulWithValidBarCode(): void
+    {
+        $rackId = 'AB12345678';
+        $fluidXPlate = new FluidXPlate($rackId);
+        self::assertSame($rackId, $fluidXPlate->rackId);
+        self::assertCount(96, $fluidXPlate->wells());
+        self::assertCount(96, $fluidXPlate->freeWells());
+        self::assertCount(0, $fluidXPlate->filledWells());
+    }
+
+    public function testCanNotAddInvalidBarcode(): void
+    {
+        $barcode = 'testWrongBarcode';
+        $this->expectExceptionObject(new TecanScanException('Invalid tube barcode:' . $barcode));
```

Comment:
Minimize the code that comes after to clarify what exactly triggers the exception